### PR TITLE
Extension altar pillar height to summon grim

### PR DESCRIPTION
**Pull request by hisacat**
From The book "Death, and How to Cure It!"
-"To summon him, you must build an altar consisting of 3 obsidian columns that are at least 2 blocks high. They may be higher if you like."

It's written with "They may be higher if you like." but **actually altar height only works with 2 heights**.

This code makes workable altar height can be higher (can edit with config.java)

I know this is a very minor issue.
but anyway, this issue can be fixed simply so I create this pr.

If you are ok then please check.

### CODE REVIEW

File: common/src/main/java/mca/server/ReaperSpawner.java [hunk 1]
```diff
@@ -136,11 +136,23 @@ private boolean isNightTime(World world) {
         return time >= 13000 && time <= 23000;
     }
 
-    private Set<BlockPos> getTotems(World world, BlockPos pos) {
-        return Stream.of(HORIZONTALS).map(d -> pos.offset(d, 3)).filter(pillarCenter ->
-                world.getBlockState(pillarCenter).isOf(Blocks.OBSIDIAN)
-                && world.getBlockState(pillarCenter.down()).isOf(Blocks.OBSIDIAN)
-                && world.getBlockState(pillarCenter.up()).isIn(BlockTags.FIRE)).map(BlockPos::up).collect(Collectors.toSet());
+    private Set<BlockPos> getTotemsFires(World world, BlockPos pos) {
+        int maxPillarHeight = Math.max(2, Config.getInstance().summonGrimReaperAltarMaxPillarHeight); // overridden to 2 if it less than 2.
```
Comment (CDAGaming) [inline common/src/main/java/mca/server/ReaperSpawner.java]: Eh, tbf you could make this instead be max between 2 and worldHeight. People wouldn't really want to configure the maximum but moreso the minimum requirements.

So optimally, this line should be Math.max(minPillarHeight, world.topY);

**Comment by CDAGaming** [comment]
@hisacat Left some comments on here, once this is adjusted, I'll look into merging this into both 1.18 branches

**Comment by hisacat** [comment]
@CDAGaming  ah, that makes sense. thanks.
now variable name changed to minPillarHeight use Math.min with world.getTopY() for getting pillars max height.
and I'd remove some comments on codes.

**Comment by hisacat** [comment]
I was an idiot. some code fixed and tested

**Pull request merged**